### PR TITLE
Fix stats on empty tracksLength vector

### PR DESCRIPTION
**Pull request by fabiencastan**
fix poparteu/openMVG#261

### CODE REVIEW

File: src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp [hunk 1]
```diff
@@ -966,6 +965,9 @@ double SequentialSfMReconstructionEngine::ComputeTracksLengthsHistogram(Histogra
     vec_nbTracks.push_back(obs.size());
   }
 
+  if (vec_nbTracks.empty())
```
Comment (simogasp) [inline src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp]: why not put the check at the very beginning?
```c++
if (_sfm_data.GetLandmarks().empty)
return -1.0;
```

Comment (simogasp) [inline src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp]: also while at it...
```c++
for(const auto &track : _sfm_data.GetLandmarks())
{
  const Observations & obs = track.second.obs;
  vec_nbTracks.push_back(obs.size());
}
```

Comment (fabiencastan) [inline src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp]: yes, it's better!

Comment (fabiencastan) [inline src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp]: done

**Pull request merged**